pkg/services/publicdashboards/api/api.go

```diff
@@ -149,6 +150,15 @@ func (api *Api) QueryPublicDashboard(c *models.ReqContext) response.Response {
 		return response.Error(http.StatusBadRequest, "invalid panel ID", err)
 	}
 
```

Comment:
Let's get the dashboard first and error if that fails for some reason

Reply:
I don't think we should do any query before validating that a path param received is a valid integer